test/geo.test.js

```diff
@@ -142,3 +144,188 @@ describe('GeoPoint', function() {
   });
 
 });
+
+describe('nearFilter()', function() {
+  it('handles a null where filter parameter gracefully', function() {
+    var result = nearFilter(null);
+    result.should.be.false;
+  });
+
+  it('handles an empty where filter parameter gracefully', function() {
+    var result = nearFilter({});
+    result.should.be.false;
+  });
+
+  it('returns a parsed near object when provided a flat where filter with location', function() {
+    var result = nearFilter({
+      location: {
+        near: { lat: 0, lng: 0 },
+      },
+    });
+
+    should.exist(result);
+    result.near.lat.should.equal(0);
+    result.near.lng.should.equal(0);
+    should.not.exist(result.maxDistance);
+    should.not.exist(result.unit);
+    result.key.length.should.be.exactly(1);
+    result.key[0].should.be.exactly('location');
```

Comment:
The assertion failures would produce rather unhelpful messages. Can we simplify to something like the following?

```js
should(result).eql({
  near: {lat: 0, lng: 0},
  key: ['location']
});
```